code/src/sixsq/nuvla/server/resources/spec/deployment.cljc

```diff
@@ -148,28 +204,13 @@
              :json-schema/order 32)))
 
 
-(s/def ::data-records-filter
-  (-> (st/spec string?)
-      (assoc :name "data-records-filter"
-             :json-schema/type "string"
-             :json-schema/editable false
-             :json-schema/indexed false
-
-             :json-schema/display-name "data records filter"
-             :json-schema/description "Filter used to fetch data-records to process"
-             :json-schema/order 33)))
-
-
-
 (def deployment-keys-spec
   (su/merge-keys-specs [common/common-attrs
                         {:req-un [::module
                                   ::state
                                   ::api-endpoint]
                          :opt-un [::api-credentials
-                                  ::data-objects            ;;deprecated
```

Comment:
@0xbase12 Do I have to put them back with full definitions behind?

Reply:
Yes, if not, old deployment documents will fail spec on edit.

Reply:
If you want, you can create an ES migration script (to be executed on upgrade of system) to update existing document to the new schema.

Reply:
I put back the removed fields. Also created https://github.com/nuvla/api-server/issues/333